File: src/main/webapp/js/recipe/recipe.controller.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

(function(){
    angular.module('app')
    .controller('recipeCtrl', recipeCtrl);

    recipeCtrl.$inject = ['$scope','Recipe','Category','Ingredient'];

    function recipeCtrl($scope, Recipe, Category, Ingredient){

        $scope.recipes = [];
        $scope.ingredients = [];
        $scope.categories = [];

        $scope.newRecipe = {
            name: '',
            description: '',
            steps: '',
            difficulty: 0,
            price: 0,
            ingredients: [],
            categories: []
        };

        function resetNewRecipe(){
            $scope.newRecipe = {
                name: '',
                description: '',
                steps: '',
                difficulty: 0,
                price: 0,
                ingredients: [],
                categories: []
            };
        }

        Recipe.query(function(recipes){
            $scope.recipes = recipes;
        });
        Category.query(function(categories){
            $scope.categories = categories;
        });
        Ingredient.query(function(ingredients){
            $scope.ingredients = ingredients;
        });

        $scope.saveNewRecipe = function(){
            Recipe.save($scope.newRecipe, function(success){
                $scope.recipes.push(success);
                resetNewRecipe();
            }, function(err){
                console.log(err);
            });
        };
    }
})();

File: src/main/java/stuff/valm0n/restful/ui/controllers/RecipeController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package stuff.valm0n.restful.ui.controllers;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import stuff.valm0n.restful.ui.models.Ingredient;
import stuff.valm0n.restful.ui.models.Recipe;
import stuff.valm0n.restful.ui.repositories.IngredientRepository;
import stuff.valm0n.restful.ui.repositories.RecipeRepository;

/**
 *
 * @author vsimon
 */
@RestController
@RequestMapping("/recipes")
public class RecipeController {
    
    @Autowired
    RecipeRepository repository;
    
    @RequestMapping(value="", method=RequestMethod.GET)
    public List<Recipe> getAllContaining(@RequestParam(value="name", defaultValue="") String name){
        return repository.findByNameContainingIgnoreCase(name);
    }
    
    @RequestMapping(value="/{id}", method=RequestMethod.GET)
    public Recipe getRecipe(@PathVariable("id") Long id){
        return repository.findById(id);
    }
    
    @RequestMapping(value="", method=RequestMethod.POST)
    public Recipe saveNewRecipe(@RequestBody Recipe recipe){
        return repository.save(recipe);
    }
    
    @RequestMapping(value="/{id}", method=RequestMethod.DELETE)
    public void deleteRecipe(@PathVariable("id") Long id){
        repository.delete(id);
    }
    
    
    
    
}
